src/AddItem.js

```diff
@@ -1,21 +1,91 @@
+import { useState, useEffect } from 'react';
+import getToken from './lib/tokens';
 import { firestore } from './lib/firebase';
+import { useHistory } from 'react-router-dom';
 
 const AddItem = () => {
+  const [item, setItem] = useState('');
+  const [frequency, setFrequency] = useState(0);
+  const [lastPurchasedDate, setLastPurchasedDate] = useState(null);
+  const history = useHistory();
+
+  // For testing purposes to make sure we can push new item to Firestore
+  useEffect(() => {
+    const token = getToken();
+    localStorage.setItem('token', JSON.stringify(token));
+  });
+
   // get a reference to the Firestore document
-  const handleClick = async () => {
+  const handleClick = async (e) => {
+    e.preventDefault();
     const itemTemplate = {
-      name: 'New item',
-      date: Date.now(),
+      name: item,
+      token: JSON.parse(localStorage.getItem('token')),
+      frequency,
+      lastPurchasedDate,
     };
     await firestore.collection('items').add(itemTemplate);
+
+    alert(`Successfully added ${item} to your list!`);
+    history.push('/list');
   };
 
-  return <button onClick={handleClick}>Add an item</button>;
+  const handleChange = (e) => {
+    if (e.target.name === 'item') {
+      setItem(e.target.value);
+    } else if (e.target.name === 'frequency') {
+      setFrequency(e.target.value);
+    }
+  };
+
+  return (
+    <div>
+      <h1>Smart Shopping List</h1>
+      <form onSubmit={handleClick}>
+        <label htmlFor="item">
```

Comment:
I don't think your labels are working as expected here. The `for` attribute on your `label` element looks for a matching `id`, which at the moment doesn't exist on your `input` elements.

You can try playing around with the sample code on the [MDN `<label>` docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label). (If the labels are connected correctly, you should be able to click on the label text and focus will move to the input. Try deleting the `id` from one of the inputs, and you'll see that clicking on that label no longer affects the checkbox.)